src/components/InputItem.tsx

```diff
@@ -1,27 +1,51 @@
 import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addNewItemCreator, checkAllItemsCreator } from '../store/actions';
+import { PropsType } from '../types/types';
 
-function InputItem(): JSX.Element {
+function InputItem({ checkAll, setCheckAll }: PropsType ): JSX.Element {
   const [value, setValue] = useState('');
-  const [checkAll, setCheckAll] = useState(false);
+
+  const dispatch = useDispatch();
+  const checkAllItems = (): void => {
+    dispatch(checkAllItemsCreator(checkAll));
+  };
+  const addNewItem = (event: React.FormEvent<HTMLFormElement>): void => {
+    event.preventDefault();
+    dispatch(addNewItemCreator(value));
+  };
 
   return (
     <div className="input">
       <input
         className="input__checkbox"
         type="checkbox"
         checked={checkAll}
+        onChange={checkAllItems}
       />
       <label
         className="input__label"
-        onClick={() => setCheckAll(!checkAll)}
-      />
-      <input
-        className="input__field"
-        type="text"
-        placeholder="What needs to be done?"
-        value={value}
-        onChange={event => setValue(event.target.value)}
+        onClick={() => {
+          setCheckAll(!checkAll);
+          checkAllItems();
+        }}
       />
+      <form onSubmit={event => {
```

Comment:
you can just write `onSubmit={addNewItem}` and move `setValue('')` to `addNewItem` function